thorlcr/activities/keyedjoin/thkeyedjoinslave.cpp

```diff
@@ -1704,11 +1777,12 @@ class CKeyedJoinSlave : public CSlaveActivity, implements IJoinProcessor, implem
         bool compressed = false;
         Owned<IEngineRowAllocator> fetchDiskAllocator;
         CThorContiguousRowBuffer prefetchSource;
-        std::vector<PartIO> partIOs;
+        //Need ptr to std::vector as std::atomic's are not constructable(doesn't have copy constructor)
+        std::unique_ptr<std::vector<std::atomic<PartIO *>>> partIOs;
```

Comment:
Does this need to change?  The previous code looks fine to me.  Is this trying to protect partIO.stream?

Reply:
Atomic on a PartIO pointer don't really help.
queryFetchPartIO is overwriting the PartIO objects (not pointers), the pointers never change after init in this version.

I think the was correct before this latest commit, using an atomic partIOCreated, which is checked by the stats call.
(since we clarified that the atomic read code is effectively fenced).